### PR TITLE
Option to remove original file that's converted

### LINKED ISSUE #3: Option to remove converted file
From lab meeting, it can help with storage usage to provide easy hook to opt into automatic removal of file that's been converted

### PR DESCRIPTION
**Pull request by vreuter**
Some stylistic stuff also included, but the main things are:
- CLI option
- quick sanity check of combo of opts/args used
- `follow` a conversion with input file removal

Close #3

### CODE REVIEW

File: sra_convert.py [hunk 1]
```diff
@@ -20,66 +27,64 @@ def _parse_cmdl(cmdl):
 			"[Default: $SRARAW:" + safe_echo("SRARAW") + "]")
 
 	parser.add_argument(
-			"-m", "--mode", default="convert",
-			help="Mode. Can be: 'convert', 'delete_sra', 'delete_bam'")
-	
+			"-m", "--mode", choices=[CONVERT, DELETE_SRA, DELETE_BAM], default=CONVERT,
+			help="Subcommand action to be performed")
```
Comment (nsheff) [inline sra_convert.py]: 'subcommand action to be performed' is a bit hard to understand...

**Comment by nsheff** [comment]
looks ok but I can't figure out where the removal is actually happening

File: sra_convert.py (hunk 1 above)
Comment (vreuter) [reply sra_convert.py]: OK will change

**Comment by vreuter** [comment]
> looks ok but I can't figure out where the removal is actually happening

It's intended to be with the `follow` action in the `pm.run` call for when `args.mode == CONVERT`. Will that not do it?

**Comment by nsheff** [comment]
Now I see it. You're right. that should work perfectly. that's exactly what `follow` is for, and I hadn't even thought of that. nice!

Just as a side note, from my perspective, while I understand the benefit of moving all the commands into variables, it did make things harder for me to understand because you had to define the function `do_rm` which is doing nothing more than seeing if `args.remove` is set... it just created a lot of abstraction layers that I think improved flexibility, and I think you intended them to make things more understandable, but I think in this case it actually hurt at least my ability to understand. I was also distracted by all those changes so I missed the important one, which was quite small :)

but anyway, I approve, this should work great, thanks!

**Review by nsheff — APPROVED** [review]

**Pull request merged**